packages/cli/src/base.ts

```diff
@@ -10,13 +10,25 @@ import chalk from 'chalk'
 import net from 'net'
 import Web3 from 'web3'
 import { getGasCurrency, getNodeUrl } from './utils/config'
-import { enumEntriesDupWithLowercase, requireNodeIsSynced } from './utils/helpers'
-export const gasOptions = {
-  auto: 'auto',
-  Auto: 'auto',
-  ...enumEntriesDupWithLowercase(Object.entries(Token)),
-  ...enumEntriesDupWithLowercase(Object.entries(StableToken)),
-}
+import { getFeeCurrencyWhitelist, requireNodeIsSynced } from './utils/helpers'
+/**
+ * TODO(Arthur): Probably need to change the `gasOption` definition because we support arbitrary
+ * addresses, and not just a set of well-defined strings (cUSD, cEUR, ...).
+ *
+ * I defined a `getGasOptions` function in `helpers.ts` that makes a contract call to fetch the
+ * whitelisted address, but I need to consider that the list of gasOptions is different per
+ * network (alfajores, mainnet, baklava), because:
+ * 1. the fee currency addresses are different per network,
+ * 2. the whitelist is different, alfajores has more addresses for internal testing purposes.
+ *
+ * Question: How do I support 'auto' if `gasOptions` will a list of accepted addresses,
+ * and not a list of strings.
+ */
+export type GasOption = 'auto' | StrongAddress
```

Comment:
Leaving a note on my current thinking before handing over to @nicolasbrugneaux: 

I'm not sure the 'auto' arguments makes a lot of sense anymore. Instead, we might want to just accept `StrongAddress`, and assume that no gas currency means that CELO should be used.

Reply:
I agree